text/0000-raw-txn-header.md

```diff
@@ -0,0 +1,105 @@
+- Feature Name: raw_txn_header
+- Start Date: 2018-08-03
+- RFC PR:
+- Sawtooth Issue:
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an additional API call, txn.get_raw_header(), which will 
+return the transaction's header bytes.
+
+# Motivation 
+[motivation]: #motivation
+
+For security reasons transaction processor would like to verify the incoming
+transaction header, and signature, if TP that runs inside SGX enclave, we 
+can't trust the signature verification done by sawtooth since it happens
+outside of SGX (TEE).
+In order to verify the transaction signature, TP needs to get the serialized
+transaction header and not just all the fields.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC adds a new method to directly access the transaction header bytes.
+Transaction Processor developers will use this API when they choose to re-verify
+the signature inside of a transaction processor.
+
+No change is recommended to the existing methods, so the implementation of this RFC
+should not cause any breakage for pre-existing transaction processors.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Calling get_raw_header() will return the serialized bytes of the transaction 
+object without the need to re-serialize.
+There is also an option to add a flag to the TP registration in order to let the 
+TP decide if it requires this API.
+
+A possible implementation approach would be to modify TpRegisterRequest to
+indicate that the transaction processor desires header bytes.
+
+    message TpRegisterRequest {
+        message TpProcessRequestHeaderStyle {
+            STYLE_UNSET,
```

Comment:
[Question]: Default is EXPANDED, do we still need STYLE_UNSET? If so, in which case?
Apologies if I am missing something here and for late commenting on it.